### PR TITLE
Workaround wip fixes for fatmach0 and strings

### PR DESCRIPTION
**Pull request by radare**

### CODE REVIEW

**Comment by radare** [comment]
rebased

File: libr/bin/file.c [hunk 1]
```diff
@@ -990,16 +998,31 @@ R_API void r_bin_file_get_strings_range(RBinFile *bf, RList *list, int min, int
 	if (min < 0) {
 		return;
 	}
+#if 0
```
Comment (ret2libc) [inline libr/bin/file.c]: please just remove this code.

File: libr/util/buf.c [hunk 2]
```diff
@@ -753,14 +751,14 @@ R_API int r_buf_read_at(RBuffer *b, ut64 addr, ut8 *buf, int len) {
 		}
 		if (real_len != len) {
 			memset (buf, b->Oxff, len);
-			len = real_len;
-			if (len < 0) {
+			if (0 && len < 0) {
```
Comment (ret2libc) [inline libr/util/buf.c]: ???

File: libr/util/buf.c [hunk 3]
```diff
@@ -753,14 +751,14 @@ R_API int r_buf_read_at(RBuffer *b, ut64 addr, ut8 *buf, int len) {
...
 				return 0;
 			}
 		}
 	}
 	// addr will be converted to pa inside r_buf_cpy
 	return r_buf_cpy (b, addr, buf, b->buf, len, false);
+// eprintf ("ER %llx  -- %02x %02x %02x %02x %02x === %d\n", addr, buf[0], buf[1], buf[2], buf[3], buf[4], (int)b->length);
```
Comment (ret2libc) [inline libr/util/buf.c]: just remove

**Comment by radare** [comment]
this pr is just a hack to get strings working for some fatmach0s, but the root issue is in the rbin api, it requires much more chnages, i just wanted it to solve some cases, but its clearly not a generic fix

**Comment by ret2libc** [comment]
So I guess we can close this so that whoever has time/will can work on a generic solution, ok?

**Comment by radare** [comment]
i prefer to keep it there for testing or understanding the problem with fat bins when someone takes a minute to discuss all my rbin refactoring plans and ideas. otherwise we cant see the whole problem

**Comment by radare** [comment]
that's not going to be merged, better keep the work in separate prs cleaning up rbin

**Pull request closed without merging**